src/main/haskell/kore/src/Kore/Variables/Sort.hs

```diff
@@ -35,15 +32,25 @@ class TermWithSortVariablesClass term var where
     sortVariables :: term -> Set.Set var
 
 instance TermWithSortVariablesClass CommonKorePattern UnifiedSortVariable where
-    sortVariables = patternBottomUpVisitor sortVarsVisitor
+    sortVariables = Recursive.fold sortVarsVisitor
       where
-        sortVarsVisitor p =
-            addPatternSortVariables p (addSortVariables asUnified) (fold p)
+        sortVarsVisitor (_ :< up) =
+            case up of
+                UnifiedMetaPattern p ->
+                    addPatternSortVariables p addUnifiedSortVariables (fold p)
+                UnifiedObjectPattern p ->
+                    addPatternSortVariables p addUnifiedSortVariables (fold p)
+        addUnifiedSortVariables
+            :: MetaOrObject level
+            => Set.Set UnifiedSortVariable
+            -> Sort level
+            -> Set.Set UnifiedSortVariable
+        addUnifiedSortVariables = addSortVariables asUnified
 
 instance TermWithSortVariablesClass CommonMetaPattern (SortVariable Meta) where
-    sortVariables = cata sortVarsVisitor
+    sortVariables = Recursive.fold sortVarsVisitor
```

Comment:
Isn't fold defined as cata?

Reply:
Yes, `fold = cata`. I think we should prefer the plain English names. I'm all for proper mathematical terminology when it exists, but `recursion-schemes` is filled with cryptic Greek terms that seem to have been conjured ad hoc for the original recursion schemes paper and that aren't used anywhere else. And while I can decipher "catamorphism" from its roots, I'm already lost by the time I get to "apomorphism," to say nothing of "zygohistomorphic prepromorphism".